packages/web-client/app/components/image-editor/index.ts

```diff
@@ -1,69 +1,151 @@
 import { action } from '@ember/object';
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
+import * as Sentry from '@sentry/browser';
 import Cropper from 'cropperjs';
 import 'cropperjs/dist/cropper.css';
-
 interface ImageEditorComponentArguments {
-  aspectRatio: number; // in px
-  image: string;
+  width: number;
+  height: number;
+  image: string; // data url or url
+  fileType: string;
   rounded: false;
-  saveImageEditData(data: { image: string }): void;
+  saveImageEditData(data: { preview: string; file: Blob }): void;
   onClose(): void;
 }
 
 export default class ImageEditorComponent extends Component<ImageEditorComponentArguments> {
   cropper: Cropper | undefined;
+  canvas: HTMLCanvasElement | undefined;
+  @tracked errored = false;
   @tracked loading = true;
-  @tracked output: string | undefined;
+  @tracked preview: string | undefined;
 
   @action initializeCropper(image: HTMLImageElement) {
-    let component = this;
-    this.cropper = new Cropper(image, {
-      autoCrop: true,
-      aspectRatio: this.args.aspectRatio,
-      autoCropArea: 0.9,
-      viewMode: 1,
-      cropBoxMovable: false,
-      cropBoxResizable: false,
-      dragMode: 'move',
-      toggleDragModeOnDblclick: false,
-      ready: function (this: HTMLImageElement & { cropper: Cropper }) {
-        const dataURL = this.cropper.getCroppedCanvas().toDataURL();
-        component.output = dataURL;
-        component.loading = false;
-      },
-      cropend(this: HTMLImageElement & { cropper: Cropper }) {
-        const dataURL = this.cropper.getCroppedCanvas().toDataURL();
-        component.output = dataURL;
-      },
-      zoom(this: HTMLImageElement & { cropper: Cropper }) {
-        const dataURL = this.cropper.getCroppedCanvas().toDataURL();
-        component.output = dataURL;
-      },
-    });
+    try {
+      if (!this.args.width || !this.args.height) {
+        throw new Error(
+          'Both width and height are required to initialize cropper'
+        );
+      }
+      this.errored = false;
+      let component = this;
+      this.cropper = new Cropper(image, {
+        autoCrop: true,
+        aspectRatio: this.args.width / this.args.height,
+        autoCropArea: 0.9,
+        viewMode: 1,
+        cropBoxMovable: false,
+        cropBoxResizable: false,
```

Comment:
I was able to use scrolling on desktop to resize the image, but I think that’s somewhat obscure and wouldn’t work for certain pointer devices, maybe this should be `true`? I know this was already `false`, I just noticed it now 🤓

Reply:
Pinch zoom/resize is possible for pointer devices. This UI is emulating Twitter's image selection + cropping - not sure if it's achieving the same effect?